CentreForDigitalHumanities/parseport
# Feature/add mg parser

This PR adds the MG Parser to ParsePort's Docker compose network.

In order to test/review this, you will need to:
- Clone the https://github.com/CentreForDigitalHumanities/mg-parser-server repo and put it in the same folder as the `parseport` project folder. (Make sure `parseport` and `mg-parser-server` are siblings.)
- Checkout out the `feature/basic-api` branch in `mg-parser-server`.
- Run the following command in the main `parseport` directory to build a network with dev containers:

```shell
docker compose --profile dev up --build -d
```

(The Angular container `pp-ng` may take a while - check its logs to see its progress.) After everything is booted up, go to `http://localhost:5000`,visit the parser at `Minimalist Parser` > `Parse sentence` and wait a few seconds.

If you see the following:

![image](https://github.com/user-attachments/assets/e206a6dd-5be1-4134-a859-f82e151b592c)

It means the MG Parser server has responded with a `{"status": "ok"}`, meaning that it is ready to receive requests!

Enter a sentence, hit 'Submit' and check the `pp-mg-parser` container logs. If you see the following (or something similar), it means your input was successfully parsed.

![image](https://github.com/user-attachments/assets/b02f7d9e-903a-464c-999a-806aeb6de46e)
